ntbs-service/Services/UserService.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using ntbs_service.Models;
+
+namespace ntbs_service.Services
+{
+  public interface IUserService
+  {
+    Task<List<TBService>> TbServices(ClaimsPrincipal user);
+    UserType GetUserType(ClaimsPrincipal user);
+  }
+
+  public class UserService : IUserService
+  {
+    private readonly AdfsOptions config;
+    private readonly NtbsContext context;
+    
+    public UserService(NtbsContext context, IOptionsMonitor<AdfsOptions> options) {
+      this.context = context;
+      config = options.CurrentValue;
+    }
+    async public Task<List<TBService>> TbServices(ClaimsPrincipal user)
+    {
+      var services = await context.TBService.ToListAsync();
+      // National Team users have accss to all services' records
+      if (GetUserType(user) == UserType.NationalTeam)
+      {
+        return services;
+      }
+
+      var roles = GetRoles(user);
+      return services.Where(service => roles.Contains(service.ServiceAdGroup))
```

Comment:
Wouldn't this be equivalent to
`service => roles.Contains(service.ServiceAdGroup) || roles.Contains(service.PHECAdGroup)`?

Reply:
For all intended purposes - yes :) Happy to change to that.